task_sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -286,13 +286,13 @@ def exit(n: int) -> NoReturn:
 
 @attrs.define()
 class WatchedSubprocess:
-    id: UUID
+    id: UUID | None
```

Comment:
`id` shouldn't be ever None though

Reply:
It was none for Dag parsing process. So I think moving these to a new subclass would be better.

Reply:
yeah both id and client were always none for the dag parsing

Reply:
client was None for dag processing; id was not None for either of them, but it was not an explicit argument which was confusing. there's no need for it to not be an explicit param.

Reply:
Yup, it was passed uuid for parsing process too:

https://github.com/apache/airflow/blob/f3fd262de274b4fd1e26d36f9cead0a56775a309/airflow/dag_processing/manager.py#L860-L870